profiler/src/main/java/com/splunk/opentelemetry/profiler/RecordingSequencer.java

```diff
@@ -54,18 +54,22 @@ public void start() {
 
   @VisibleForTesting
   void handleInterval() {
-    if (!recordingEscapeHatch.jfrCanContinue()) {
-      logger.warn("JFR recordings cannot proceed.");
-      if (recorder.isStarted()) {
-        recorder.stop();
+    try {
+      if (!recordingEscapeHatch.jfrCanContinue()) {
+        logger.warn("JFR recordings cannot proceed.");
+        if (recorder.isStarted()) {
+          recorder.stop();
+        }
+        return;
       }
-      return;
-    }
-    if (!recorder.isStarted()) {
-      recorder.start();
-      return;
+      if (!recorder.isStarted()) {
+        recorder.start();
+        return;
+      }
+      recorder.flushSnapshot();
+    } catch (Throwable throwable) {
```

Comment:
Can we catch a `RuntimeException` here? Or is there a need to catch `Error`s too?

Reply:
I guess this is open for debate how we should handle it. Anything that gets thrown out of this method break profiling. Would it be ok for it to break when there is an `OOM`? If not then `RuntimeException` wouldn't be enough. I personally would lean towards catching everything as when this thing breaks it will be completely impossible to debug, best to let it never break and log any unexpected exceptions.

Reply:
I mostly thought about `ThreadDeath`, but OOM is also a valid scenario too. Honestly I'd normally expect OOMs to break absolutely everything, but in case of profiling maybe it's worth to just try to ignore them and gather as much info as possible? @breedx-splk

Reply:
Yeah, I thought the general rule for this was to almost never catch `Throwable` but rather catch `Exception`. Errors are almost never recoverable and trying to keep going after an `Error` has been thrown is going to be counterproductive (making a bad situation worse).

My recommendation is to catch `Exception` here.

Reply:
In a regular application I have never needed to catch a `Throwable`. With javaagnet the thing is that you are running inside someone else's application which is pretty much a hostile territory. You don't know whether their app sometimes ooms. If that happens inside your code then in my opinion it is really good if it is recoverable as otherwise if it ends up as support case it will be really hard to figure out why suff sometimes breaks.

Reply:
Ok I think you convinced me. I'm ok with it.

Reply:
@johnbley

Reply:
Catching `Throwable` is sometimes OK.  For example, instrumentation might accidentally attempt to allocate a 120 GB array,  causing an "OOM" - the instrumentation can abort, leaving the application unfazed.  Now, you might say, "how in the world could we distinguish between this and a "regular" OOM that is actually the app reaching its limits due to, e.g., a memory leak in the app itself?  If we swallow the OOM the app might not get notified."  Well, if the app is actually near its limit, there will be another OOM coming soon enough.
`ThreadDeath` absolutely cannot be swallowed.  I am annoyed that the current ByteBuddy instrumentation doesn't have a clean way to do the truly paranoid thing which is
```java
try {
   instrumentationStuff()
} catch (ThreadDeath td) {
  throw td;
} catch (Throwable t) {
  try {
      logProblem(t);
  } catch (ThreadDeath td) {
     throw td;
  } catch (Throwable t2) {
        // swallow 
  }
}
```
This pattern hopefully only needs to be in a few very high-level entry points to instrumentation.

Reply:
Why do you care about `ThreadDeath` that much? Isn't it only used by `Thread.stop` which has been deprecated for like 20 years. Are there really applications that try to stop random threads?

Reply:
Yes, it's deprecated; yes, developers should not use it; yes, there are not that many that do.  But!  With an extra half dozen lines of code, our instrumentation will not break the small number of apps/systems that do use it.  The first rule of instrumentation is "do no harm".

Reply:
As far as I can tell it would only break if user code for some unknown reason decided to find our profiler periodic executor thread which is named `JFR Recording Sequencer` call `stop` on it and then `join` and hang because thread wasn't really stopped. I could understand if this catch Throwable was inserted somewhere in user code and could suppress death of a user thread, but this is inside our agent code. Is this truly something we should worry about? If suppressing `ThreadDeath` is bad then wouldn't swallowing of `InterruptedException` also be bad?

Reply:
@johnbley @laurit @breedx-splk can we merge this PR?

Reply:
@mateuszrzeszutek you and @iNikem have to decide whether to catch `Throwable` or `RuntimeException` or something else

Reply:
Lauri's argument convinced me and I am OK with catching `Throwable` here.

Reply:
I'm fine with that too - nobody's gonna call Thread.stop() on our profiler thread.

Reply:
Okay, there are no objections - I'm hitting the merge button then